### PR TITLE
[Altair] Add check for missed EF tests

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

I've added a programmatic method for checking that we accessed all the files in the `eth2.0-spec-tests` dir whenever we run `make test-ef`.

This method feels a bit yucky, you have to remember to register each file as you access it in the tests (but that's not *too* painful IMO). However, given that there are multiple varieties of tests all running on different threads, this was the best way I could come up with.

The most interesting part of this PR is that it thinks we aren't running the rewards tests, that's why this CI run should fail.

I deleted the `tests/mainnet/altair/rewards` directory and our tests still pass, indicating to me that we're indeed not running these tests. Perhaps that's a TODO/FIXME I haven't managed to find. Do you have thoughts @michaelsproul? 

## Additional Info

NA

### CODE REVIEW

**Comment by michaelsproul** [comment]
The missing `rewards` tests are interesting.

Spec: https://github.com/ethereum/eth2.0-specs/tree/v1.1.0-alpha.4/tests/formats/rewards

They mostly test phase0 rewards functions like `get_source_deltas` which Lighthouse doesn't quite have an equivalent of due to our use of a single-pass optimisation in `get_attestation_deltas`:

https://github.com/sigp/lighthouse/blob/967d680a9801bcb83676f8cdb6825b23580148cc/consensus/state_processing/src/per_epoch_processing/base/rewards_and_penalties.rs#L67-L124

Rather than calculating all the deltas individually in multiple iterations we calculate them all in one iteration, combining them as we go.

To run the bulk of the `rewards` tests, we'd need to refactor that function into one that outputs all the delta vecs, and then applies them. I think this wouldn't be too intrusive a change, and will increase our coverage on phase0 functions before the Altair fork (still ~3 months away).

The function `get_inactivity_penalty_deltas` remains relevant in Altair, so we should test it regardless.

**Comment by michaelsproul** [comment]
https://github.com/sigp/lighthouse/pull/2349 is ready for review, and then we can merge this

**Comment by paulhauner** [comment]
@michaelsproul this is ready for review/merge :)

File: testing/ef_tests/check_all_files_accessed.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+excluded_paths = [
+    # Things from future phases
+    "tests/mainnet/config/custody_game.yaml",
+    "tests/mainnet/config/sharding.yaml",
+    "tests/mainnet/config/merge.yaml",
+    "tests/minimal/config/custody_game.yaml",
+    "tests/minimal/config/sharding.yaml",
+    "tests/minimal/config/merge.yaml",
+    # Genesis Validity
+    "tests/minimal/altair/genesis/validity",
```
Comment (michaelsproul) [inline testing/ef_tests/check_all_files_accessed.py:30]: I didn't realise we were missing these (or I'd forgotten). Will look into it tomorrow.

**Comment by michaelsproul** [comment]
The diff looked super sus on this PR (it included all of #2349 even though it was already merged?). However it seemed to merge down correctly: https://github.com/sigp/lighthouse/pull/2279/commits/97d1807ed12565debb0a00774e2928089b0928f0

**Pull request merged**